fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java

```diff
@@ -1888,6 +1893,70 @@ public void testDirectRelationshipsOk() throws IOException {
+        setAuth(postParent, "fedoraAdmin");
+        final String parentUri;
+        try (final CloseableHttpResponse response = execute(postParent)) {
+            assertEquals(CREATED.getStatusCode(), getStatus(response));
+            parentUri = getLocation(response);
+        }
+        // Make parent only accessible to fedoraAdmin
+        final String parentAcl = "@prefix acl: <http://www.w3.org/ns/auth/acl#> .\n" +
+                "<#readauthz> a acl:Authorization ;\n" +
+                "   acl:agent \"fedoraAdmin\" ;\n" +
+                "   acl:mode acl:Read, acl:Write ;\n" +
+                "   acl:accessTo <" + parentUri + "> .";
+        ingestAclString(parentUri, parentAcl, "fedoraAdmin");
+        // Admin can see parent
+        final HttpGet getAdminParent = getObjMethod(parent);
+        setAuth(getAdminParent, "fedoraAdmin");
+        assertEquals(OK.getStatusCode(), getStatus(getAdminParent));
+        final HttpGet getParent = getObjMethod(parent);
+        setAuth(getParent, username);
+        // testUser92 cannot see parent.
+        assertEquals(FORBIDDEN.getStatusCode(), getStatus(getParent));
+
+        final String child = getRandomUniqueId();
+        final HttpPost postChild = postObjMethod(parent);
+        postChild.setHeader("Slug", child);
+        setAuth(postChild, "fedoraAdmin");
+        final String childUri;
+        try (final CloseableHttpResponse response = execute(postChild)) {
+            assertEquals(CREATED.getStatusCode(), getStatus(response));
+            childUri = getLocation(response);
+        }
+        // Make child accessible to testUser92
+        final String childAcl = "@prefix acl: <http://www.w3.org/ns/auth/acl#> .\n" +
+                "<#readauthz> a acl:Authorization ;\n" +
+                "   acl:agent \"" + username + "\" ;\n" +
+                "   acl:mode acl:Read, acl:Write ;\n" +
+                "   acl:accessTo <" + childUri + "> .";
+        ingestAclString(childUri, childAcl, "fedoraAdmin");
+        // Admin can see child.
+        final HttpGet getAdminChild = getObjMethod(parent + "/" + child);
+        setAuth(getAdminChild, "fedoraAdmin");
+        assertEquals(OK.getStatusCode(), getStatus(getAdminChild));
+
+        // testUser92 can see child.
+        final HttpGet getChild = getObjMethod(parent + "/" + child);
+        setAuth(getChild, username);
+        assertEquals(OK.getStatusCode(), getStatus(getChild));
+
+        // Admin bypasses ACL resolution gets 409.
+        final HttpGet getAdminRequest = getObjMethod(parent + "//" + child);
+        setAuth(getAdminRequest, "fedoraAdmin");
+        assertEquals(BAD_REQUEST.getStatusCode(), getStatus(getAdminRequest));
+        // User can't access the URL (because the path is not correct).
+        final HttpGet getUserRequest = getObjMethod(parent + "//" + child);
+        setAuth(getUserRequest, username);
+        assertEquals(FORBIDDEN.getStatusCode(), getStatus(getUserRequest));
```

Comment:
This is better than a stacktrace... I am fine with 403 for the 5.1.1 patch release.

Reply:
Yeah, I didn't look to closely to figure out where the empty path exception comes from, but I feel like it comes from Modeshape. I was able to perform the matching with this
```
    private boolean hasEmptyPathSegments(final HttpServletRequest httpRequest) {
        final String requestPath = httpRequest.getContextPath() + httpRequest.getServletPath() +
                httpRequest.getRequestURI();
        final String finalTestPath;
        if (requestPath.startsWith("/") && requestPath.endsWith("/") && requestPath.length() > 1) {
            finalTestPath = requestPath.substring(1, requestPath.length() - 2);
        } else if (requestPath.startsWith("/")) {
            finalTestPath = requestPath.substring(1);
        } else if (requestPath.endsWith("/")) {
            finalTestPath = requestPath.substring(0, requestPath.length() - 1);
        } else {
            finalTestPath = requestPath;
        }
        if (finalTestPath.contains("/")) {
            final String[] paths = finalTestPath.split("/");
            if (Arrays.stream(paths).anyMatch(String::isEmpty)) {
                return true;
            }
        }
        return false;
    }
```
I'm not sure how fragile that is, but if you want I can add a commit here to return the appropriate 400 Bad Request. I'd probably add more test cases here than in the 6.0.0 location.

Reply:
Thanks, @whikloj . However, I am inclined to move forward with this PR as it stands.

Reply:
Okay, so you want we to revert the last commit?

Reply:
No... it is an improvement. Thanks.